Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -454,6 +476,49 @@ public struct TargetSourcesBuilder {
 
         return contents
     }
+
+    public static func computeContents(for generatedFiles: [AbsolutePath], toolsVersion: ToolsVersion, additionalFileRules: [FileRuleDescription], defaultLocalization: String?, targetName: String, targetPath: AbsolutePath, observabilityScope: ObservabilityScope) -> (sources: [AbsolutePath], resources: [Resource]) {
+        var sources = [AbsolutePath]()
+        var resources = [Resource]()
+
+        generatedFiles.forEach { absPath in
+            // 5.6 handled treated all generated files as sources.
+            if toolsVersion <= .v5_6 {
+                sources.append(absPath)
+                return
+            }
+
+            var rule = Self.computeRule(for: absPath, toolsVersion: toolsVersion, additionalFileRules: additionalFileRules, observabilityScope: observabilityScope)
+
+            // If we did not find a rule for a generated file, we treat it as to be processed for now. Eventually, we should handle generated files the same as other files and require explicit handling in the manifest for unknown types.
+            if rule == .none {
+                rule = .processResource(localization: .none)
+            }
+
+            switch rule {
+            case .compile:
+                if absPath.extension == "swift" {
+                    sources.append(absPath)
+                } else {
+                    observabilityScope.emit(warning: "Generated plugin source files can only use Swift: \(absPath)")
```

Comment:
Nit: Maybe "Only Swift is supported for generated plugin source files at this time" to be similar to the other messages here?